apps/etterlatte-saksbehandling-ui/client/src/components/behandling/soeknadsoversikt/familieforhold/Familieforhold.tsx

```diff
@@ -1,68 +1,27 @@
+import { FamilieforholdOmstillingsstoenad } from '~components/behandling/soeknadsoversikt/familieforhold/omstillingsstoenad/FamilieforholdOmstillingsstoenad'
 import { Heading } from '@navikt/ds-react'
-import { GjenlevendeForelder } from './personer/GjenlevendeForelder'
-import { Barn } from './personer/Barn'
 import { ContentHeader } from '~shared/styled'
-import { Border, DashedBorder } from '../styled'
-import { SoeskenListe } from './personer/Soesken'
-import { GyldigFramsattType, IDetaljertBehandling, IGyldighetproving } from '~shared/types/IDetaljertBehandling'
-import { VurderingsResultat } from '~shared/types/VurderingsResultat'
-import { Foreldre } from '~components/behandling/soeknadsoversikt/familieforhold/personer/Foreldre'
 
 export interface PropsFamilieforhold {
   behandling: IDetaljertBehandling
 }
 
 export const Familieforhold: React.FC<PropsFamilieforhold> = ({ behandling }) => {
-  const innsenderErGjenlevende =
-    behandling.gyldighetsprøving?.vurderinger.find(
-      (g: IGyldighetproving) => g.navn === GyldigFramsattType.INNSENDER_ER_FORELDER
-    )?.resultat === VurderingsResultat.OPPFYLT
-
-  if (behandling.familieforhold == null || behandling.søker == null) {
-    return <div style={{ color: 'red' }}>Familieforhold kan ikke hentes ut</div>
-  }
-
-  const doedsdato = behandling.familieforhold.avdoede.opplysning.doedsdato
-
   return (
     <>
-      {behandling.gyldighetsprøving?.resultat === VurderingsResultat.OPPFYLT ? (
-        <>
-          <ContentHeader>
-            <Heading spacing size="medium" level="2">
-              Familieforhold
-            </Heading>
-          </ContentHeader>
-          <FamilieforholdWrapper>
-            <Barn person={behandling.søker} doedsdato={doedsdato} />
-            <DashedBorder />
-            <Foreldre
-              gjenlevende={behandling.familieforhold.gjenlevende.opplysning}
-              innsenderErGjenlevende={innsenderErGjenlevende}
-              doedsdato={doedsdato}
-              avdoed={behandling.familieforhold.avdoede.opplysning}
-            />
-            <DashedBorder />
-            <SoeskenListe soekerFnr={behandling.søker.foedselsnummer} familieforhold={behandling.familieforhold!!} />
-          </FamilieforholdWrapper>
-        </>
+      <ContentHeader>
+        <Heading spacing size="medium" level="2">
+          Familieforhold
+        </Heading>
+      </ContentHeader>
+      {behandling.sakType === ISaksType.BARNEPENSJON ? (
+        <FamilieforholdBarnepensjon behandling={behandling} />
       ) : (
-        <>
-          <FamilieforholdWrapper>
-            <GjenlevendeForelder
-              person={behandling.familieforhold?.gjenlevende.opplysning}
-              innsenderErGjenlevendeForelder={innsenderErGjenlevende}
-              doedsdato={doedsdato}
-            />
-          </FamilieforholdWrapper>
-        </>
+        <FamilieforholdOmstillingsstoenad behandling={behandling} />
```

Comment:
Finnes det noen måte å gjøre dette mer eksplisitt? Her gjør vi bare antagelsen av at dersom det ikke er barnepensjon så må det være omstillingsstønad. Altså en slags `when` som passer på at alle verdiene i `ISakType` er gjort rede for. 

Dette funker, men er kanskje ikke så pent? 
```
{
        {
          [ISaksType.BARNEPENSJON]: <FamilieforholdBarnepensjon behandling={behandling} />,
          [ISaksType.OMSTILLINGSSTOENAD]: <FamilieforholdOmstillingsstoenad behandling={behandling} />,
        }[behandling.sakType]
      }
```

Reply:
Kan se på det i en egen PR da en evt. endring vil gjelde flere steder i koden 😄